src/examples/button2/button2.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"machine"
+	"time"
+)
+
+// This example assumes that you are using the pca10040 board
+
+func setLED(led *machine.GPIO, state bool) {
+	if state {
+		led.High()
+	} else {
+		led.Low()
+	}
+}
+
+func main() {
+	led1 := machine.GPIO{machine.LED1}
+	led1.Configure(machine.GPIOConfig{Mode: machine.GPIO_OUTPUT})
+
+	led2 := machine.GPIO{machine.LED2}
+	led2.Configure(machine.GPIOConfig{Mode: machine.GPIO_OUTPUT})
+
+	led3 := machine.GPIO{machine.LED3}
+	led3.Configure(machine.GPIOConfig{Mode: machine.GPIO_OUTPUT})
+
+	led4 := machine.GPIO{machine.LED4}
+	led4.Configure(machine.GPIOConfig{Mode: machine.GPIO_OUTPUT})
+
+	button1 := machine.GPIO{machine.BUTTON1}
+	button1.Configure(machine.GPIOConfig{Mode: machine.GPIO_INPUT_PULLUP})
+
+	button2 := machine.GPIO{machine.BUTTON2}
+	button2.Configure(machine.GPIOConfig{Mode: machine.GPIO_INPUT_PULLUP})
+
+	button3 := machine.GPIO{machine.BUTTON3}
+	button3.Configure(machine.GPIOConfig{Mode: machine.GPIO_INPUT_PULLUP})
+
+	button4 := machine.GPIO{machine.BUTTON4}
+	button4.Configure(machine.GPIOConfig{Mode: machine.GPIO_INPUT_PULLUP})
+
+	for {
+		setLED(&led1, button1.Get())
```

Comment:
You could also do `led1.Set(button1.Get())`. I think it's a little cleaner. What do you think?

Reply:
I actually implemented as a separate function just to test passing a pointer to a struct. But you are of course quite correct.

Reply:
Such tests are currently mostly done in examples/test, adding such a test here seems like the wrong place to me.

I would like to add a real testsuite at some point for such tests but that hasn't been implemented yet. I'd like to do that together with Travis CI setup which requires an easier way to install LLVM (possible after the LLVM 7 release a few days ago).